probe-rs/src/debug/debug_info.rs

```diff
@@ -809,6 +810,13 @@ impl DebugInfo {
                                 .and_then(|register| register.value);
                             match reg_val {
                                 Some(reg_val) => {
+                                    if reg_val.is_zero() {
+                                        // If we encounter this rule for CFA, it implies the scenario depends on a FP/frame pointer to continue successfully.
+                                        // Therefore, if reg_val is zero (i.e. FP is zero), then we do not have enough information to determine the CFA by rule.
+                                        stack_frames.push(return_frame);
```

Comment:
Are you sure we cannot continue and just drop this one frame?

Reply:
Pretty sure. The CFA is required to calculate the LR value, and if the CFA is invalid, then the LR is invalid, and we don't know what the next frame is. 

I have compared results against gdb/openocd and we end up with the same frames in the unwind.